### ISSUE TITLE
When can this tool support Chinese servers?

### ISSUE THREAD

**Issue by shangguansb** [razaqq/PotatoAlert#24]
When can this tool support Chinese servers? This is really a great tool, thanks to the developers.

**Comment by shangguansb** [comment]
Or can you provide the java or python language version of this tool? I can develop java language or python language

**Comment by razaqq** [comment]
As far as I know there EU, NA, RU and ASIA servers. I have never heard of a chinese server.
The WoWs API which this entire tool relies on only has those servers too, so no clue what you mean.

Also this is only the frontend, the entire backend is written in Go.

**Comment by shangguansb** [comment]
First of all, thank you very much. This is the user record query URL of the Chinese server: https://wowsgame.cn/zh-cn/community/accounts/search?search=%E6%97%A0%E6%95%8C%E8 %88%B0%E9%98%9F
You can try to open it. After querying this page, the data in XML format will be returned. I hope you can support the Chinese server in your plug-in. Thank you again, you are really a very good people~

**Comment by razaqq** [comment]
that wont work, i need a REST api like this https://developers.wargaming.net/reference/all/wot/account/list/?r_realm=ru

**Comment by shangguansb** [comment]
Understand, I think I can write code in GO language. I have 5 years of back-end development experience in Alibaba. You can upload the back-end code to github, and then I use GO to write the data conversion part of the code.

**Comment by razaqq** [comment]
There are two problems i have with your idea:
1. That site is EXTREMELY slow here in europe, its impossible that it could keep up with the traffic
2. The backend would require MAJOR restructuring which im not sure i want to do.

What i would instead suggest is to try to find the endpoints used by the site.
You can sniff around on that site and find a bunch of undocumented api endpoints like
- `https://vortex.wowsgame.cn/api/encyclopedia/zh_cn/`
- `http://vortex.wowsgame.cn/api/accounts/7049627625/` where `7049627625` is the account id of the player
- `http://vortex.wowsgame.cn/api/accounts/7049627625/ships/`
- `http://vortex.wowsgame.cn/api/accounts/7049627625/clans/`

What I could not find was the following:
- Find account id by player name, this has to exist tho since the site can find them
- Some ship specific stats like average damage (usually from `https://api.worldofwarships.com/wows/ships/stats/`), this has to exist tho since the site can display it

It should be possible to find the rest of them aswell with something like fiddler and proxifier.

**Comment by shangguansb** [comment]
ok  thanks,i  got it

**Comment by shangguansb** [comment]
I found an API to query user id by user name,
https://wowsgame.cn/zh-cn/community/accounts/search/?search=zhanjia&pjax=1,
But it is not a rest Api.

**Comment by razaqq** [comment]
you can get something usable with

```python
import requests

response = requests.get("https://wowsgame.cn/zh-cn/community/accounts/autocomplete/", params={"query": "zhanjia"}, headers={"X-Requested-With": "XMLHttpRequest"})
print(response.json())
```

to get

```
{
  'query': 'zhanjia',
  'suggestions': [
    {
      'data': 7051804109,
      'value': 'ZHANJIAM  FGD'
    },
    {
      'data': 7053570594,
      'value': 'ZHANJIAN1'
    },
    {
      'data': 7050769212,
      'value': 'zhanjian2020'
    },
    {
      'data': 7051330667,
      'value': 'zhanjian777888'
    },
    {
      'data': 7051283914,
      'value': 'zhanjian8888'
    },
    {
      'data': 7051443460,
      'value': 'zhanjian_20'
    },
    {
      'data': 7051418135,
      'value': 'ZHANJIAN_2020'
    },
    {
      'data': 7051462707,
      'value': 'zhanjian_妖妖'
    },
    {
      'data': 7051306734,
      'value': 'zhanjianshijie'
    },
    {
      'data': 7051817482,
      'value': 'zhanjianzhiwan'
    }
  ]
}
```

i will do some more advancted search with fiddler and proxifier later today

**Comment by shangguansb** [comment]
Yes, the technology of the Chinese server operator is relatively backward, and there is no public release of the api...

**Comment by razaqq** [comment]
ill finish the next update of the client and then start implementing this in the backend

**Comment by shangguansb** [comment]
thank u  ！！！！

**Comment by shangguansb** [comment]
Hi , have you  update  PotatoAlert for the china server?  the  latst veersion is?

**Comment by razaqq** [comment]
No i havent finished the 3.1.0 update (https://github.com/razaqq/PotatoAlert/tree/3.1.0) yet, only then will i start with smth like this

**Comment by shangguansb** [comment]
Great, looking forward to your work~

**Comment by shangguansb** [comment]
So, when do you plan to update the version that runs through the server?

**Comment by razaqq** [comment]
as i previously said, first i want to finisht the 3.1.0 update, then i will have a look at this
if this was as simple as adding a few api endpoints, then i would have done it already
the chinese server simply doesnt have a public api so this is only possible by major restructuring of the backend server

**Comment by shangguansb** [comment]
Approximately when can you expect to start?

**Comment by razaqq** [comment]
As i said previously: when the next update is finished. I only work on this when i have the time and motivation to do so, im not getting paid.

If you want it faster, you can write a backend implementation for the chinese server yourself. Since this project is entirely open source you can see the api here yourself https://github.com/razaqq/PotatoAlert/blob/3.1.0/Client/StatsParser.cpp

If you provide me with a working and tested implementation i am happy to merge it with the existing one.
Another option is that i give you access to the backend code and you fork and work on it directly, your call.

I simply dont have the motivation to implement some hacky solution for the chinese server specifically, because its quite frankly not my fault they dont have a proper working api.

**Comment by razaqq** [comment]
I need two things for this: a) a replay played on the chinese server b) the `preferences.xml` from game files from the chinese client (there is a tag like `<active_server>	WOWS EU	</active_server>`) in there and i need to know how the tag looks like for the chinese server

**Comment by HenryQuan** [comment]
a), [20220819_191907_PBSB002-Warspite-1941_37_Ridge.zip](https://github.com/razaqq/PotatoAlert/files/9606647/20220819_191907_PBSB002-Warspite-1941_37_Ridge.zip), this is a recent replay. I have larger replays if needed.
b), it is `<active_server> WOWS 360 </active_server>`. 360 is name of the current partner company with Wargaming. It is known for its security software on Windows.

For the game folder detection, the key is always different. It is using `WOWS.CN.PRODUCTION` instead of `WOWS.WW.PRODUCTION`. There is also a seperate gamecenter for it. Now, this may introduce a slight problem. I have both the global client and the Chinese client installed on the machine. How to know which one is running?
